### PR TITLE
Add deploy doc

### PR DESCRIPTION
**Pull request by thedavidprice**
This doc sets a foundation for information about deploy targets in general. Most importantly, it covers Vercel deployment which is ready to go.

### CODE REVIEW

File: docs/deploy.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Deploy
+
+Redwood is designed for serverless architecture deployment, which requires a unique continuous deployment process:
+
+1. code is committed to a repository on GitHub, GitLab, or Bitbucket, which triggers the deployment
```
Comment (jtoar) [inline docs/deploy.md:5]: the which clause needs to be closer to the action (code is commited)

**Comment by thedavidprice** [comment]
Mockup of iteration:
- increased h1 to 2.25rem
- increased h2 to 1.75rem
- retain IA structure -- both visually (for scanning) and nav
- using text and underline to differentiate hide/reveal (could go with one or other)

Was specifically struck by how changes to header size improved ability to scan structure visually.

Screen Shot 2020-08-18 at 1 32 01 PM

File: docs/deploy.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+The `url` setting above accesses the database connection string via an environment variable, `DATABASE_URL`. Using env vars is the recommended method for both ease of development process as well as security best practices.
+
+### Environment Variables
+Any environment variables used locally, e.g. in your `env.defaults` or `.env`, must also be added to your hosting provider settings. (See documentation specific to your provider.)
+
+Additionally, if your application uses env vars on the Web Side, you must configure Redwood's build process to make them available in production. See the [Redwood Environment Variables doc](https://redwoodjs.com/docs/environment-variables) for instructions.
+
+## Netlify
+
+To deploy Redwood on Netlify, see the [Tutorial Deployment section](https://deploy-preview-295--redwoodjs.netlify.app/tutorial/deployment).
```
Comment (jtoar) [inline docs/deploy.md:50]: (Is it that we deploy _to_ platforms?)
```suggestion
To deploy Redwood to Netlify, see the [Tutorial Deployment section](https://deploy-preview-295--redwoodjs.netlify.app/tutorial/deployment).
```

File: docs/deploy.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+If you simply want to experience the Netlify deployment process without a database and/or adding custom code, you can do the following:
+1. create a new redwood project: `yarn create redwood-app ./netlify-deploy`
+2. after your "netlify-deploy" project installation is complete, init git, commit, and add it as a new repo to GitHub, BitBucket, or GitLab
+3. follow the applicable steps in the deployment section of the Tutorial (see link above)
```
Comment (jtoar) [inline docs/deploy.md:55]:
```suggestion
3. follow the applicable steps in the deployment section of the Tutorial (see the link above)
```

File: docs/deploy.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+## Vercel
+_The following instructions assume you have read the [General Deployment Setup](#general-deployment-setup) section above._
+
+### Project Init
+If you haven't already, we recommend experiencing the full Redwood DX via the [Redwood Tutorial](https://redwoodjs.com/tutorial/welcome-to-redwood). Simply return to these instructions when you come to the Deployment section.
+
+If you simply want to experience the Vercel deployment process without a database and/or adding custom code, you can do the following:
+1. create a new redwood project: `yarn create redwood-app ./vercel-deploy`
+2. after your "vercel-deploy" project installation is complete, init git, commit, and add it as a new repo to GitHub, BitBucket, or GitLab
```
Comment (jtoar) [inline docs/deploy.md:65]: Make instructions a little more explicit:
```suggestion
2. after your "vercel-deploy" project installation is complete, `cd` into the directory, initialize a git repository (`git init`), make your first commit (`git commit -am 'first'`), and add it as a new repo to GitHub, BitBucket, or GitLab
```

File: docs/deploy.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+Additionally, if your application uses env vars on the Web Side, you must configure Redwood's build process to make them available in production. See the [Redwood Environment Variables doc](https://redwoodjs.com/docs/environment-variables) for instructions.
+
+## Netlify
+
+To deploy Redwood on Netlify, see the [Tutorial Deployment section](https://deploy-preview-295--redwoodjs.netlify.app/tutorial/deployment).
+
+If you simply want to experience the Netlify deployment process without a database and/or adding custom code, you can do the following:
+1. create a new redwood project: `yarn create redwood-app ./netlify-deploy`
+2. after your "netlify-deploy" project installation is complete, init git, commit, and add it as a new repo to GitHub, BitBucket, or GitLab
```
Comment (jtoar) [inline docs/deploy.md:54]: Make instructions a little more explicit:
```suggestion
2. after your "netlify-deploy" project installation is complete, `cd` into the directory, initialize a git repository (`git init`), make your first commit (`git commit -am 'first'`), and add it as a new repo to GitHub, BitBucket, or GitLab

```

**Comment by thedavidprice** [comment]
@jtoar Thanks! I was working locally on some updates and went through each of yours that way. Huge help 🚀

**Pull request merged**